### DISCUSSION TITLE
I have a few of questions about Oqtane

### DISCUSSION THREAD

**Discussion by vnetonline** [oqtane/oqtane.framework#3214] [Q&A]
@sbwalker 

1. What is the difference between Tenant and Site ... can there be multiple tenants ?
2. What is the relationship between Users and the Site how is a user linked to the Site ... I can't see SiteId in the Users table hence I ask.

**Comment by sbwalker** [comment]
Oqtane is multi-tenant. It supports 2 tenancy models: Shared and Isolated. Internally to the framework, a "tenant" is synonymous with a database. Therefore "tenants" are physically isolated from one another (ie. different databases). Tenants contain one or more "sites". Sites are logically separated from one another within the same tenant (via a SiteId in the database) - so this allows for shared tenancy. Note that since Oqtane uses standard auto-incrementing id columns in its database architecture, it means that a specific Site within an installation must be identified by a combination of its TenantId and SiteId. All external resources which need to support multi-tenancy must take this into consideration ie. file system, cache keys, etc...

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/7890e3dc-5c51-4a56-b6e7-efd89042499b)

Users are scoped to a Tenant (a database). This allows them to be shared across Sites within a Tenant. They are mapped to a specific Site based on their Role assignments (ie. the Registered User role).

**Reply by vnetonline** [reply]
@sbwalker So is there a UI for creating new tenants and mapping them to isolated databases ??

**Reply by sbwalker** [reply]
@vnetonline this has been covered in numerous presentations and demos... including the first video in the module development video series (https://youtu.be/JPfUZPlRRCE?si=hRKcCzGYoquKAmcR)

- Login as Host
- display Control Panel
- choose Admin Dashboard 
- choose Site Management
- choose Add Site
- Database - choose < Create New Database >

This will create the site in a new database/tenant (as I mentioned above, tenants are synonymous with databases).

**Reply by vnetonline** [reply]
Thanks @sbwalker I will give it a go
